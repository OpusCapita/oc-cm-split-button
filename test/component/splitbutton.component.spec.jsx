/* eslint-disable no-unused-expressions */
import React from 'react';
import renderer from 'react-test-renderer';
import SplitButton from '../../src/splitbutton.component';
import Icon from '../../src/icon.component';

const menuItems = [
  {
    id: 'item_id_21',
    title: 'Item 1',
    onClick: () => {}, // es
  },
  {
    id: 'item_id_22',
    title: 'Item 2',
    onClick: () => {},
  },
  {
    id: 'item_id_d1',
    type: 'divider',
  },
  {
    id: 'item_id_23',
    title: 'Item 3',
    onClick: () => {},
    disableClosing: true,
  },
];

describe('SplitButton', () => {
  test('renders correctly (enabled)', () => {
    const splitButton = renderer
      .create(
        <SplitButton
          id="split-button"
          title="Split Button"
          icon={
            <Icon
              faIcon="fa fa-check"
              aria-hidden="true"
              color="lightgreen"
              size="1.5rem"
              margin="0 0.5rem 0 0"
            />
          }
          disabled={false}
          onButtonClick={() => {}}
          menuDisabled={false}
          menuItems={menuItems}
        />,
      )
      .toJSON();
    expect(splitButton).toMatchSnapshot();
  });

  test('renders correctly (disabled)', () => {
    const splitButton = renderer
      .create(
        <SplitButton
          id="split-button"
          title="Split Button"
          icon={
            <Icon
              faIcon="fa fa-check"
              aria-hidden="true"
              color="lightgreen"
              size="1.5rem"
              margin="0 0.5rem 0 0"
            />
          }
          disabled={false}
          onButtonClick={() => {}}
          menuDisabled={false}
          menuItems={menuItems}
        />,
      )
      .toJSON();
    expect(splitButton).toMatchSnapshot();
  });
});
