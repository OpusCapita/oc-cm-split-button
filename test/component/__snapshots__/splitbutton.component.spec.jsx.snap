// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplitButton renders correctly (disabled) 1`] = `
<section
  className="sc-kEYyzF kmXTDj"
  id="split-button"
>
  <button
    className="sc-bwzfXH IsBR"
    disabled={false}
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="sc-hMqMXs fa fa-check krqEzY"
      color="lightgreen"
      size="1.5rem"
    />
    Split Button
  </button>
  <div
    className="oc-dropdown-menu"
  >
    <div
      className="sc-kkGfuU brzToE dropdown btn-group"
      primary={0}
      style={
        Object {
          "bsSize": "xs",
          "bsStyle": "info",
        }
      }
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="dropdown-toggle btn btn-default"
        disabled={false}
        id="split-button-dropdown"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        type="button"
      >
        <i
          className="sc-hMqMXs fa fa-angle-down qPVAu"
          color="white"
          size="2rem"
        />
      </button>
      <ul
        aria-labelledby="split-button-dropdown"
        className="dropdown-menu dropdown-menu-right"
        role="menu"
      >
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            id="item_id_21"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            <span
              className="oc-dropdown-menu-icon"
            />
            <span
              className="oc-dropdown-menu-title"
            >
              Item 1
            </span>
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            id="item_id_22"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            <span
              className="oc-dropdown-menu-icon"
            />
            <span
              className="oc-dropdown-menu-title"
            >
              Item 2
            </span>
          </a>
        </li>
        <li
          className="divider"
          onKeyDown={[Function]}
          role="separator"
        />
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            id="item_id_23"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            <span
              className="oc-dropdown-menu-icon"
            />
            <span
              className="oc-dropdown-menu-title"
            >
              Item 3
            </span>
          </a>
        </li>
      </ul>
    </div>
  </div>
</section>
`;

exports[`SplitButton renders correctly (enabled) 1`] = `
<section
  className="sc-kEYyzF kmXTDj"
  id="split-button"
>
  <button
    className="sc-bwzfXH IsBR"
    disabled={false}
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="sc-hMqMXs fa fa-check krqEzY"
      color="lightgreen"
      size="1.5rem"
    />
    Split Button
  </button>
  <div
    className="oc-dropdown-menu"
  >
    <div
      className="sc-kkGfuU brzToE dropdown btn-group"
      primary={0}
      style={
        Object {
          "bsSize": "xs",
          "bsStyle": "info",
        }
      }
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="dropdown-toggle btn btn-default"
        disabled={false}
        id="split-button-dropdown"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        type="button"
      >
        <i
          className="sc-hMqMXs fa fa-angle-down qPVAu"
          color="white"
          size="2rem"
        />
      </button>
      <ul
        aria-labelledby="split-button-dropdown"
        className="dropdown-menu dropdown-menu-right"
        role="menu"
      >
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            id="item_id_21"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            <span
              className="oc-dropdown-menu-icon"
            />
            <span
              className="oc-dropdown-menu-title"
            >
              Item 1
            </span>
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            id="item_id_22"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            <span
              className="oc-dropdown-menu-icon"
            />
            <span
              className="oc-dropdown-menu-title"
            >
              Item 2
            </span>
          </a>
        </li>
        <li
          className="divider"
          onKeyDown={[Function]}
          role="separator"
        />
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            id="item_id_23"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            <span
              className="oc-dropdown-menu-icon"
            />
            <span
              className="oc-dropdown-menu-title"
            >
              Item 3
            </span>
          </a>
        </li>
      </ul>
    </div>
  </div>
</section>
`;
